### PR TITLE
updated inputs to use HOC instead of reusing same input

### PR DESCRIPTION
**Pull request by alastairmcqueen**
## Description

Brief description of changes made e.g. bug fix for broken Ajax container.

## Related tickets

Any Jira tickets related to the changes.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

- [ ] I have lint my changes locally.
- [ ] I have formatted my changes locally.
- [ ] I have updated any relevant documentation.
- [ ] I have added tests to cover my changes.
- [ ] I have checked that all new and existing tests passed.
- [ ] I have checked to ensure there aren't other open Pull Requests for the same update/change.

### CODE REVIEW

File: src/components/select/index.js [hunk 1]
```diff
@@ -1,7 +1,7 @@
 import React from "react";
-import InputField from "../input-field";
+import Input from "../input";
 import withAutocomplete from "../with-auto-complete";
 
-const Select = props => <InputField {...props} fullWidth />;
+const Select = props => <Input {...props} fullWidth noRef />;
```
[the remark this answers is not in the document]
Comment (alastairmcqueen) [reply src/components/select/index.js:5]: It's a component, I assure you the label is there

**Review by MikeyAlmighty — APPROVED** [review]

File: src/components/select/index.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (MikeyAlmighty) [reply src/components/select/index.js:5]: @accesslint fight me

File: src/components/with-input-wrapper/styles.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
+import styled from "styled-components";
+import Flex from "../flex";
+import { colors, fontSizes, fontWeights, space } from "../theme";
+
+export const StyledContainer = styled(Flex).attrs(() => ({
```
Comment (Pdut89) [inline src/components/with-input-wrapper/styles.js:5]: [dust] Can use shorthand:
```
export const StyledContainer = styled(Flex).attrs({
  flexDirection: 'column'
})`
```

**Review by Pdut89 — APPROVED** [review]

File: src/components/time-input/index.js [hunk 3]
```diff
@@ -1,58 +1,6 @@
...
-};
-
-TimeInput.propTypes = defaultPropTypes;
-
-TimeInput.defaultProps = {
-  label: "",
-  inputStyle: {},
-  theme: theme
-};
+const TimeInput = props => <Input {...props} type="time" step="1" />;
```
Comment (GaryGeorgeu) [inline src/components/time-input/index.js:4]: Not sure if this `type="time"` is handled on the component itself using a custom time selector, or if you are using the standard `type="time"` attribute for an input. 

If you aren't using a custom time selector component, then just note that the `type="time"` isn't natively supported in safari, IE, or edge.
https://caniuse.com/input-datetime

**Review by GaryGeorgeu — APPROVED** [review]

File: src/components/time-input/index.js (hunk 3 above)
Comment (alastairmcqueen) [reply src/components/time-input/index.js:4]: @GaryGeorgeu good catch, i'll have to implement something to fix that

File: src/components/with-input-wrapper/styles.js (hunk 2 above)
Comment (tiaanduplessis) [reply src/components/with-input-wrapper/styles.js:5]: Agreed. I was mistaken previously regarding this syntax being removed. It appears only to be the case for subfunctions https://github.com/styled-components/styled-components/blob/master/CHANGELOG.md#v500---2020-01-13

File: src/components/with-auto-complete/index.js [hunk 4]
```diff
@@ -9,130 +9,113 @@ import {
...
-      </StyledContainer>
-    );
+  const handleListItemClicked = ({ value, label }) => {
+    setValue(name, value);
+    adjustValues();
+    onChange({ label, value });
   };
 
-  return AutoComplete;
+  const ListOptions = () =>
```
Comment (tiaanduplessis) [inline src/components/with-auto-complete/index.js:86]: Can we try and move these components out of render?

File: src/components/time-input/index.js (hunk 3 above)
Comment (tiaanduplessis) [reply src/components/time-input/index.js:4]: You should be able to add a polyfill, tho quality varies

**Review by tiaanduplessis — CHANGES_REQUESTED** [review]

File: src/components/with-auto-complete/index.js (hunk 4 above)
Comment (alastairmcqueen) [reply src/components/with-auto-complete/index.js:86]: Will give it a go

**Review by mantonides — APPROVED** [review]

**Pull request merged**